src/util/PathUtil.ts

```diff
@@ -1,48 +1,28 @@
 import { posix, win32 } from 'path';
+import escapeStringRegexp from 'escape-string-regexp';
 import urljoin from 'url-join';
 import type { TargetExtractor } from '../http/input/identifier/TargetExtractor';
 import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
 import type { HttpRequest } from '../server/HttpRequest';
 import { BadRequestHttpError } from './errors/BadRequestHttpError';
 
-// Characters to ignore when URL decoding the URI path to a file path.
-const pathComponentDelimiters = [ '/', '\\' ];
-// Regex to find all instances of encoded path component delimiters.
-const encodedDelimiterRegex = new RegExp(
-  `${pathComponentDelimiters.map((delimiter): string => encodeURIComponent(delimiter)).join('|')}`, 'giu',
+// URI path component delimiters and their escaped versions
+const pathComponentDelimiters = [ '/', '\\' ]
+  .flatMap((delimiter): string[] => [
+    delimiter,
+    encodeURIComponent(delimiter),
+    encodeURIComponent(delimiter).toLowerCase(),
+    encodeURIComponent(encodeURIComponent(delimiter)),
+    encodeURIComponent(encodeURIComponent(delimiter)).toLowerCase(),
```

Comment:
If I understand correctly we also need these because otherwise the URL `/a%252Fb` would become the file path `/a%2Fb` which would then become the URL `/a%2Fb` right? (Would not have thought of that one myself 😅 ).

Would add some extra comments explaining why we do this here (because I see this being a cause of confusion in the future) and some extra integration tests specifically for the case we are trying to prevent here.

Reply:
But is this the way to solve this potential issue though? If the target URL is `/a%25252Fb`, it will be decoded to file path `/a%252Fb`, which will become URL `/a%252Fb` right? So the problem just shifted one level of encoding. Perhaps instead we need a specific regex to cover this case instead of storing the first few levels of encoding. Unless I'm missing something in my assumption?

Reply:
> If I understand correctly we also need these because otherwise the URL /a%252Fb would become the file path /a%2Fb which would then become the URL /a%2Fb right?

Indeed! We need to make sure a bijection exists (bar encoding casing).

> (Would not have thought of that one myself 😅 ).

Me neither, it was @bourgeoa!

> Would add some extra comments explaining why we do this here

I will try… but I can't guarantee they will be more simple than the code 😄

> it will be decoded to file path /a%252Fb, which will become URL /a%252Fb right? So the problem just shifted one level of encoding. Perhaps instead we need a specific regex to cover this case instead of storing the first few levels of encoding. 

OMG yes. I thought the first 2 levels were sufficient, but my tests were incomplete.